Distribution/Server/Features/PackageInfoJSON/State.hs

```diff
@@ -0,0 +1,251 @@
+{-# LANGUAGE DeriveDataTypeable  #-}
+{-# LANGUAGE DeriveGeneric       #-}
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell     #-}
+{-# LANGUAGE TypeFamilies        #-}
+
+module Distribution.Server.Features.PackageInfoJSON.State where
+
+import           Control.Arrow        (first, second)
+import           Control.Applicative  ((<|>))
+import           Control.Monad.Reader (ask, asks)
+import qualified Control.Monad.State  as State
+import qualified Data.Aeson           as Aeson
+import           Data.Aeson           ((.=), (.:))
+import           Data.Acid            (Query, Update, makeAcidic)
+import qualified Data.HashMap.Strict  as HashMap
+import qualified Data.Map             as Map
```

Comment:
Also it might make sense to use Data.Map.Strict here.

Reply:
```
Note: You should use Data.Map.Strict instead of this module if:

You will eventually need all the values stored.
The stored values don't represent large virtual data structures to be lazily computed.
```
Sounds like me! :+1: